rocketpy/AeroSurface.py

```diff
@@ -219,10 +240,85 @@ def kind(self, value):
             )
             area = np.pi * self.baseRadius**2
             self.k = 1 - volume / (area * self.length)
+            self.y_nosecone = Function(
+                lambda x: np.sqrt(rho**2 - (min(x - self.length, 0)) ** 2)
+                + (self.baseRadius - rho)
+            )
+
         elif value == "elliptical":
             self.k = 1 / 3
+            self.y_nosecone = Function(
+                lambda x: self.baseRadius
+                * np.sqrt(1 - ((x - self.length) / self.length) ** 2)
+            )
+
+        elif value == "vonkarman":
+            self.k = 0.5
+            theta = lambda x: np.arccos(1 - 2 * max(min(x / self.length, 1), 0))
+            self.y_nosecone = Function(
+                lambda x: self.baseRadius
+                * (theta(x) - np.sin(2 * theta(x)) / 2) ** (0.5)
+                / (np.pi**0.5)
+            )
+        elif value == "parabolic":
+            self.k = 0.5
+            self.y_nosecone = Function(
+                lambda x: self.baseRadius
+                * ((2 * x / self.length - (x / self.length) ** 2) / (2 - 1))
+            )
+
         else:
-            self.k = 0.5  # Parabolic and Von Karman
+            raise ValueError(
+                f"Nose Cone kind '{self.kind}' not found, "
+                + "please use one of the following Nose Cone kinds:"
+                + '\n\t"conical"'
+                + '\n\t"ogive"'
+                + '\n\t"lvhaack"'
+                + '\n\t"tangent"'
+                + '\n\t"vonkarman"'
+                + '\n\t"elliptical"'
+                + '\n\t"parabolic"\n'
+            )
+
+        n = 127  # Points on the final curve.
+        p = 3  # Density modifier. Greater n makes more points closer to 0. n=1 -> points equally spaced.
+
+        # Finds the tangential intersection point between the circle and nosecone curve.
+        yPrimeNosecone = lambda x: self.y_nosecone.differentiate(x)
+        xIntercept = lambda x: x + self.y_nosecone(x) * yPrimeNosecone(x)
+        radius = lambda x: (self.y_nosecone(x) ** 2 + (x - xIntercept(x)) ** 2) ** 0.5
+        xInit = fsolve(
+            lambda x: radius(x) - self.bluffiness * self.baseRadius if x > 3e-7 else 0,
+            self.bluffiness * self.baseRadius,
+            xtol=1e-7,
+        )[0]
+
+        # Corrects circle radius if it's too small.
+        if xInit > 0:
+            r = self.bluffiness * self.baseRadius
+        else:
+            r = 0
+            print(
+                "ATTENTION: The chosen bluffiness ratio is insufficient for the selected nosecone category, thereby the effective bluffiness will be 0."
```

Comment:
I suggest changing to "WARNING" instead os "ATTENTION".

However, this message is currently being raised everytime that the user do not define the bluffiness value, which may be annoying sometimes.

Can we think a better approach to raise this warning only when the user specify a bluffness value without it being valid enough?

Reply:
Done.